graphix/generator.py

```diff
@@ -9,7 +9,7 @@
 from graphix.gflow import flow, gflow, get_layers, find_odd_neighbor
 
 
-def generate_from_graph(graph, angles, inputs, outputs, timeout=100):
+def generate_from_graph(graph, angles, inputs, outputs, meas_planes):
```

Comment:
I'm still wondering whether it is good to require meas_planes for things like generate_from_graph. Most literatures on MBQC (say papers and reviews by Raussendorf) deal only with XY-plane measurement and most people may not go further than that (gflow, local-Clifford decoration, etc). I guess for those using only XY-plane it's just one-liner to create the meas_plane dict, so perhaps not very cumbersome. BTW, do we need meas_plane argument for `flow`? I see it's to raise error but I feel it's a bit strange to add it just to throw errors?
what's your thoughts @pafloxy ?

Reply:
>BTW, do we need meas_plane argument for flow? I see it's to raise error but I feel it's a bit strange to add it just to throw errors?
 
The causal flow algorithm has not been adapted to handle different measurement planes, and most probably it is not possible either given measuring in some planes would require more than one element in their correction set. So it seems we may not really need it, but maybe @masato-fuk had something on his mind ?

  >Most literatures on MBQC (say papers and reviews by Raussendorf) deal only with XY-plane measurement and most people may not go further than that (gflow, local-Clifford decoration, etc).

I agree, actually measurement patterns with other a generalised set of planes is something that has not really been considered widely before. However note that  [1] shows, even if we start with multiple measurement patterns we can alway use certain graph operations such that we are left with only measurement planes in {XY, YZ}, along side restrictions to the geometry of nodes with YZ measurements. 
Nonetheless I think it would be by no means superflous to allow all three measuremnt planes in the patterns, perhaps it might have some application we have not seen yet. 

Sorry I was bit late yesterday, I still have to check rest of the commit. But on the outset it seems pretty neat and precise. Nice work @masa10-f  ;)

Reply:
Thank you for your review!

> I'm still wondering whether it is good to require meas_planes for things like generate_from_graph. Most literatures on MBQC (say papers and reviews by Raussendorf) deal only with XY-plane measurement and most people may not go further than that (gflow, local-Clifford decoration, etc). I guess for those using only XY-plane it's just one-liner to create the meas_plane dict, so perhaps not very cumbersome. 

I agree with your opinion. I will eliminate the need for the `meas_planes` argument of some functions that use flow/gflow inside.

> BTW, do we need meas_plane argument for flow? I see it's to raise error but I feel it's a bit strange to add it just to throw errors?

In my opinion, yes. I think causal flow is a property of an underlying graph (graph with measurement planes), not a graph itself. Requiring the `meas_planes` is not meant to raise an error. An underlying graph with either XZ or YZ measurement planes actually does not have any flow from the definition. I mean that 'no flow' is also a correct answer.

How do you think about it? @pafloxy

Reply:
Thank you for your review! @shinich1 

> I'm still wondering whether it is good to require meas_planes for things like generate_from_graph. Most literatures on MBQC (say papers and reviews by Raussendorf) deal only with XY-plane measurement and most people may not go further than that (gflow, local-Clifford decoration, etc). I guess for those using only XY-plane it's just one-liner to create the meas_plane dict, so perhaps not very cumbersome. 

I agree with your opinion. I will eliminate the need for the meas_planes argument in some functions that uses flow/gflow inside.

> BTW, do we need meas_plane argument for flow? I see it's to raise error but I feel it's a bit strange to add it just to throw errors?

In my opinion, yes. I think causal flow is a property of an underlying graph (graph with measurement planes), not a graph itself. Requiring meas_planes is not meant to raise an error. An underlying graph with either XZ or YZ measurement planes actually does not have a flow. I mean that "no flow" is also a correct answer. However, as @shinich1 mentioned, in some literature flow is defined without measurement planes because they restrict all measurement planes to "XY". So, it might be good idea not to require meas_planes in `flow` for those not familiar with multiple measurement planes

Reply:
So, I will remain a conditional branch in `flow` but eliminate the need to set the `meas_planes` argument